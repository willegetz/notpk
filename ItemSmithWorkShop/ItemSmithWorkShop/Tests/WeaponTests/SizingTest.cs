using ItemSmithWorkShop.AdventureItems.MagicWeaponUtilities;
using ItemSmithWorkShop.AdventureItems.MundaneWeaponUtilites;
using ItemSmithWorkShop.WeaponUtilities;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using ApprovalTests;
using ItemSmithWorkShop.AdventureItems.ExtraordinaryWeaponUtilities;

namespace ItemSmithWorkShop.Tests.WeaponTests
{
	[TestClass]
	public class SizingTest
	{
		[TestMethod]
		public void TestSmallSizeWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			WeaponItem sizedWeapon = WeaponItemSmith.SizeWeapon(weapon, "Small");
			Approvals.Approve(sizedWeapon.DisplaySizedWeapon());
		}

		[TestMethod]
		public void TestLargeSizeWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			WeaponItem sizedWeapon = WeaponItemSmith.SizeWeapon(weapon, "Large");
			Approvals.Approve(DisplayUtilities.BasicDisplay(sizedWeapon));
			//Approvals.Approve(sizedWeapon.DisplaySizedWeapon());
		}

		[TestMethod]
		public void TestSmallAdamantineWeapon()
		{
			var weapon = WeaponItemSmith.OrderItem("Dagger");
			var sizedWeapon = WeaponItemSmith.SizeWeapon(weapon, "Small");
			var adamantineWeapon = WeaponItemSmith.OrderSpecialComponent(sizedWeapon, "Adamantine");
			Approvals.Approve(DisplayUtilities.BasicDisplay(adamantineWeapon));
		}

		[TestMethod]
		public void TestLargePlus3Weapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			WeaponItem sizedWeapon = WeaponItemSmith.SizeWeapon(weapon, "Large");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(sizedWeapon, 3);
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(magicWeapon));
		}

		[TestMethod]
		public void TestLargePlus3FlamingWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			WeaponItem sizedWeapon = WeaponItemSmith.SizeWeapon(weapon, "Large");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(sizedWeapon, 3);
			EnchantedWeaponItem enchantedWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Flaming");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(enchantedWeapon));
		}

		[TestMethod]
		public void TestLargePlus3FlamingIcyBurstWeapon()
		{
			WeaponItem weapon = WeaponItemSmith.OrderItem("Dagger");
			WeaponItem sizedWeapon = WeaponItemSmith.SizeWeapon(weapon, "Large");
			MagicWeaponItem magicWeapon = new MagicWeaponItem(sizedWeapon, 3);
			EnchantedWeaponItem flamingWeapon = WeaponEnchanter.RequestEnchantment(magicWeapon, "Flaming");
			EnchantedWeaponItem icyBurstWeapon = WeaponEnchanter.RequestEnchantment(flamingWeapon, "Icy Burst");
			Approvals.Approve(DisplayUtilities.FullMagicalDisplay(icyBurstWeapon));
		}
	}
}
