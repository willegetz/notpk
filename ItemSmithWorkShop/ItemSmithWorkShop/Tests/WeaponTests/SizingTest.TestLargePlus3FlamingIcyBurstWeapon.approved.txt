+3 Dagger [Large], Flaming, Icy Burst:	'72304 gp'	('+6 value')
To Hit: '+3'
Damage: '1d6 +3 +1D6 [19-20 x2] +1d10' Piercing or Slashing, Magic, Fire, Cold
Hardness: '32'
Hit Points: '64'
Weight: '2 pound(s)'
	The dagger is a common secondary weapon. You get a +2 bonus on
	Sleight of Hand checks made to conceal a dagger on your body.

	Upon command, a flaming weapon is sheathed in fire.
	The fire does not harm the wielder. The effect
	remains until another command is given.

	An icy burst weapon functions as a frost weapon that also
	explodes with frost upon striking a successful critical hit.
	The frost does not harm the wielder. In addition to the extra
	damage from the frost ability, an icy burst weapon deals an
	extra 1d10 points of cold damage on a successful critical hit.
	If the weapon’s critical multiplier is ×3, add an extra 2d10
	points of cold damage instead, and if the multiplier is ×4,
	add an extra 3d10 points. Bows, crossbows, and slings so
	crafted bestow the cold energy upon their ammunition.
	Even if the frost ability is not active, the weapon still
	deals its extra cold damage on a successful critical hit.

Creator Caster Level: '10'
Time to Create: '72 Days'
Creation XP Cost: '2880'
Creation Raw Material Cost: '36304'

	Required Feats: Craft Magic Arms and Armor
	Required Spells: flame blade, flame strike, or fireball
	Required Spells: chill metal or ice storm
